packages/truffle/test/scenarios/compile/compile.js

```diff
@@ -328,8 +329,14 @@ describe("Compilation with db enabled", async () => {
   let config, project;
   const logger = new MemoryLogger();
 
-  function checkForDb(config) {
-    const dbPath = path.join(config.working_directory, ".db");
+  function checkForDb() {
+    const configStore = new Configstore(
+      "truffle",
+      {},
+      { globalConfigPath: true }
+    );
+    const truffleDataDirectory = path.dirname(configStore.path);
+    const dbPath = path.join(truffleDataDirectory, ".db");
```

Comment:
We could use `TruffleConfig.getDefaultDataDirectory()` here instead, in order to express more directly the idea that the behavior should match Truffle's user-level config.

Reply:
This makes sense, so `@truffle/config` will become the `devDependency` and not `configstore`.

Reply:
Hmm, maybe we should just get rid of the special `configstore` logic and use `TruffleConfig` in `@truffle/db` as well. It is either `@truffle/config` or `configstore`.